### PR TITLE
Adapt for use with UI5 Build and Development Tooling

### PR DESCRIPTION
**Pull request by RandomByte**
OpenUI5 Sample App using the [UI5 Build and Development Tooling](https://github.com/SAP/ui5-tooling)

**Do not merge yet.** Wait until the tooling is available on npm.

### CODE REVIEW

**Comment by RandomByte** [comment]
Tooling and UI5 deps now available on npm. Testing still missing

**Comment by RandomByte** [comment]
Added tests ✅ 🎉

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Don't use package-lock.json to always have the latest dependencies
+package-lock=false
```
Comment (RandomByte) [inline .npmrc]: This also disables the use of `npm audit`:
```
❯ npm audit                                                   
npm ERR! code EAUDITNOLOCK
npm ERR! audit Neither npm-shrinkwrap.json nor package-lock.json found: Cannot audit a project without a lockfile
```

Comment (RandomByte) [inline .npmrc]: This also disables the use of `npm audit`:
```
❯ npm audit                                                   
npm ERR! code EAUDITNOLOCK
npm ERR! audit Neither npm-shrinkwrap.json nor package-lock.json found: Cannot audit a project without a lockfile
```

File: package.json [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
-    "bower": "^1.8.0",
-    "grunt": "^1.0.1",
-    "grunt-contrib-clean": "^1.0.0",
-    "grunt-contrib-connect": "^1.0.2",
-    "grunt-contrib-copy": "^1.0.0",
-    "grunt-eslint": "^20.0.0",
-    "grunt-karma": "^1.0.0",
-    "grunt-openui5": "^0.12.0",
-    "karma": "^2.0.0",
+    "@ui5/cli": "0.0.1",
```
Comment (RandomByte) [inline package.json]: Should we add this here or treat it as a general prerequisite (like grunt-cli in the past)?

And if we add it here, why not ranged: `^0.0.1`?

File: README.md [hunk 3]
```diff
@@ -1,34 +1,129 @@
...
+    npm test
+    ```
+## Building
+### Option 1: Standard preload build
+1. Execute the build
+    ```sh
+    ui5 build -a
+    ```
+1. Run the result
+    1. Install an HTTP server like [zeit/serve](https://www.npmjs.com/package/serve) (**Note:** You can use any HTTP server)
```
Comment (RandomByte) [inline README.md]: Due to a refactoring, in the latest versions of the `zeit/serve` module, no gzip is being applied anymore.

Details: https://zeit.co/blog/serve-7

Last known version with gzip support: `serve@6.5.3`

We should adapt our recommendation here to ensure a server with gzip support is being used.

Comment (matz3) [reply README.md]: Reason for not having gzip in zeit/serve: https://github.com/zeit/serve-handler/issues/1#issuecomment-390632443

File: package.json [hunk 4]
```diff
@@ -1,30 +1,33 @@
...
-    "bower": "^1.8.0",
-    "grunt": "^1.0.1",
-    "grunt-contrib-clean": "^1.0.0",
-    "grunt-contrib-connect": "^1.0.2",
-    "grunt-contrib-copy": "^1.0.0",
-    "grunt-eslint": "^20.0.0",
-    "grunt-karma": "^1.0.0",
-    "grunt-openui5": "^0.12.0",
-    "karma": "^2.0.0",
+    "@ui5/cli": "^0.1.0",
```
Comment (RandomByte) [inline package.json]: We need to update or delete the ui5 cli dependency. See https://github.com/SAP/ui5-tooling/issues/58

Comment (RandomByte) [reply package.json]: Let's update this to "^0.2.1"

@domianer @thodd

File: README.md [hunk 5]
```diff
@@ -67,7 +67,7 @@
     1. Install an HTTP server like [zeit/serve](https://www.npmjs.com/package/serve) (if not already done)
         ```sh
         # Install zeit/serve
```
Comment (RandomByte) [inline README.md]: Enhance comment to point out that version 6 must be used to have gzip available

File: Gruntfile.js [hunk 6]
```diff
@@ -0,0 +1,212 @@
+'use strict';
```
Comment (RandomByte) [inline Gruntfile.js]: This file needs to be removed

Comment (RandomByte) [reply Gruntfile.js]: @domianer

File: README.md [hunk 7]
```diff
@@ -1,34 +1,129 @@
...
     ```sh
     npm install
     ```
 
-## Usage
-### Server
-Run `grunt serve` to start a local server with your application at [http://localhost:8080](http://localhost:8080).
+1. Start a local server and run the application (http://localhost:8080/index.html)
+    ```sh
+    ui5 serve -o /index.html
```
[the remark this answers is not in the document]
Comment (RandomByte) [reply README.md]: Done

**Comment by RandomByte** [comment]
**Note:** Do not delete the `ui5-tooling` branch! Several documents link directly to this branch.

**Pull request merged**